### DISCUSSION TITLE
Platform check not equals not working as expected (multiple OR values)

### DISCUSSION THREAD

**Discussion by ktenvregelaar** [featurehub-io/featurehub#1147] [General]
**Describe the bug**
The **not** equals check on platform in a split targeting rule always applies if you select more than 1 platform.

 Which area does this issue belong to?

* [*] FeatureHub Admin Web app
* [*] SDK
* [ ] SDK examples
* [ ] Documentation
* [ ] Other

**To Reproduce**
1. Go to 'Features'
2. Create a boolean feature flag with any name
3. Set default to off
4. Add a split targeting rule with the following options:
- Platform **not equals** android or ios
- Set this split targeting rule to On
5. Use the featurehub api to check if the feature is enabled and include the platform in the request (Using server side evaluation)
6. It doesn't matter which platform you use in the request, it will always return that the feature is enabled. (android/ios/linux/windows/macos)

**Expected behavior**
If a split targeting rule is configured as Platform **not** equals android or ios I would expect it to return true if the platform is linux, windows or macos and return false if the platform is android or ios

**Versions**
 - FeatureHub version [e.g. 1.5.10]

**Comment by IrinaSouth** [comment]
Hi @ktenvregelaar, thanks for reporting the issue. 

We tested this scenario and it turns out that this is working as expected for the "not equals" operator. Reason is because those are OR conditions. And so the logic behind is  - walk through all of the values, and if any of them are not equal to the supplied value, then return true. However, we realised that we are missing "includes" and "excludes" operators, that should solve your problem and allow OR operator with what you expected. We will add this in the next release. As a workaround for now, you can use "custom" strategy, which you can setup like in the attached screen. Then in your code you can do the following (JS SDK): 
`await FeatureHub.context.attribute_value('attr_platform', 'windows').build();`

<img width="1113" alt="Screenshot 2024-04-04 at 19 10 48" src="https://github.com/featurehub-io/featurehub/assets/9538697/de95e788-855a-4365-8567-81feb297c261">

Please make sure that you do not name your Custom key as "Platform" (I used attr_platform in the above example), otherwise it will clash with the preset Platform rule.

**Comment by IrinaSouth** [comment]
Opened #1141

**Comment by IrinaSouth** [comment]
#1141  Has been resolved and will be released in the next version.
